### PR TITLE
Add a FieldButton

### PR DESCRIPTION
**Pull request by balderdash**
button

Adding a new `Field` subtype that puts a button on a block. It accepts a callback that will run when clicked. I just drew a slightly smaller rounded rectangle over a normal field input, but I think it looks clickable enough. It accepts either a string label or arbitrary svg, in case you need to fiddle with fonts or colors (e.g. if you want to use FontAwesome to put a pin icon on your button).

I'm using this to implement the location picker:

image

### CODE REVIEW

**Comment by balderdash** [comment]
[GLJ: Coordinate picker on sprite placement](https://trello.com/c/uqCRXP2y/86-glj-coordinate-picker-on-sprite-placement)

File: core/ui/fields/field_button.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+var INNER_HEIGHT = 17;
+var Y_OFFSET = -13;
+
+/**
+ * Class for a button field.
+ * @param {string} value The initial value of the field.
+ * @param {Function} buttonHandler A function that is executed when the button
+ *   is clicked. It should eventually call FieldButton.setValue() to update the
+ *   field value.
+ * @param {string} Button fill color
```
Comment (Hamms) [inline core/ui/fields/field_button.js]: nit: s/Button/color

File: core/ui/fields/field_button.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  if (this.value_ !== value) {
+    if (this.changeHandler_) {
+      // Call any change handler, and allow it to override.
+      var override = this.changeHandler_(value);
+      if (override !== undefined) {
+        value = override;
+      }
+    }
+    this.value_ = value;
+    //this.checkElement_.style.display = newState ? 'block' : 'none';
```
Comment (Hamms) [inline core/ui/fields/field_button.js]: nit: stray comment

File: core/ui/fields/field_button.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ * @param {Function} buttonHandler A function that is executed when the button
+ *   is clicked. It should eventually call FieldButton.setValue() to update the
+ *   field value.
+ * @param {string} Button fill color
+ * @param {Function} opt_changeHandler A function that is executed when the
+ *   value is changed
+ * @extends {Blockly.Field}
+ * @constructor
+ */
+Blockly.FieldButton = function(title, buttonHandler, color, opt_changeHandler) {
```
Comment (Hamms) [inline core/ui/fields/field_button.js]: should buttonHandler and color also be optional parameters?

File: tests/playground_requires.js [hunk 4]
```diff
@@ -117,3 +117,24 @@ Blockly.Blocks.sprite_variables_set = {
...
+  init: function() {
+    this.setHSV(131, 0.64, 0.62);
+    var span = document.createElementNS("http://www.w3.org/2000/svg", 'tspan');
+    span.style.fill = 'blue';
+    span.textContent = 'button';
+    this.appendDummyInput()
+        .appendTitle("here's a button")
+        .appendTitle(
+          new Blockly.FieldButton(span, function () {
+              return new Promise(resolve => resolve(prompt()));
```
Comment (joshlory) [inline tests/playground_requires.js]: I don't think this gets transpiled?  Probably fine, sine most of our dev environments support ES6.

**Pull request merged**